### PR TITLE
Update Dockerfiles for 7.0.200 and remove `PublishSingleFile`

### PR DESCRIPTION
**Pull request by richlander**
I took the opportunity to clean up our Dockerfiles, including applying the `PublishRelease` feature (new in 7.0.200).

Replaces (most of): #4441
For: #4271

### CODE REVIEW

**Review by mthalman — COMMENTED** [review]
This only applies the change to the published aspnet sample. Anyone building any of the other sample slim Dockerfiles will run into the same issue this is intended to address. Shouldn't it be applied across the board for all slim Dockerfiles for both `aspnetapp` and `dotnetapp`?

**Comment by richlander** [comment]
That's a good point. I can do that.

File: samples/README.md [hunk 1]
```diff
@@ -1,36 +1,40 @@
...
 * [Discover licensing for Linux image contents](https://github.com/dotnet/dotnet-docker/blob/main/documentation/image-artifact-details.md)
 
 ## Hosting guidance
 
-* [Host ASP.NET Core Images with Docker and HTTPS](host-aspnetcore-https.md)
-* [Push Docker Images to Azure Container Registry](push-image-to-acr.md)
-* [Push Docker Images to Docker Hub](push-image-to-dockerhub.md)
+* [Host ASP.NET Core Images with container and HTTPS](host-aspnetcore-https.md)
+* [Push container Images to Azure Container Registry](push-image-to-acr.md)
+* [Push container Images to Docker Hub](push-image-to-dockerhub.md)
```
Comment (mthalman) [inline samples/README.md:28]: Nit: lowercase `images`

**Review by mthalman — COMMENTED** [review]
Update this line to fix the failing test:
https://github.com/dotnet/dotnet-docker/blob/f4786b8c0b4469f7eb18f891fd6c090561e50006/tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs#L41

**Review by MichaelSimons — APPROVED** [review]
I like the simplification.

File: tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs [hunk 2]
```diff
@@ -36,14 +36,9 @@ public static IEnumerable<object[]> GetImageData()
 
         [DotNetTheory]
         [MemberData(nameof(GetImageData))]
-        public async Task VerifyDotnetSample(SampleImageData imageData)
+        public Task VerifyDotnetSample(SampleImageData imageData)
         {
-            if (imageData.DockerfileSuffix == "windowsservercore-iis-x64")
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs:41]: This shouldn't be deleted. Rather, it should be updated to reflect the new filenames (i.e. `windowsservercore-iis`).

File: tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs [hunk 3]
```diff
@@ -36,13 +36,13 @@ public static IEnumerable<object[]> GetImageData()
 
         [DotNetTheory]
         [MemberData(nameof(GetImageData))]
-        public async Task VerifyDotnetSample(SampleImageData imageData)
+       public async Task VerifyDotnetSample(SampleImageData imageData)
```
Comment (mthalman) [inline tests/Microsoft.DotNet.Docker.Tests/SampleImageTests.cs:39]: nit
```suggestion
        public async Task VerifyDotnetSample(SampleImageData imageData)
```

**Review by mthalman — APPROVED** [review]

**Pull request merged**